flang/lib/Lower/IntrinsicCall.cpp

```diff
@@ -1152,25 +1152,25 @@ mlir::Value IntrinsicLibrary::genIAnd(mlir::Type resultType,
 mlir::Value IntrinsicLibrary::genIchar(mlir::Type resultType,
                                        llvm::ArrayRef<mlir::Value> args) {
   // There can be an optional kind in second argument.
-  assert(args.size() >= 1);
+  assert(args.size() >= 1 && args.size() <= 2);
```

Comment:
Is there a nice way to do something like

assert(auto size = args.size(), size >=1 && size <= 2)

Reply:
That might work. Not sure how portable it would be though.

Reply:
Declare size() as a pure function :)

Reply:
https://llvm.org/doxygen/ArrayRef_8h_source.html#l00156

It's already an inline const getter. A decent C++ compiler should be able to figure it out.